### PR TITLE
DDEV: added cronjob

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

Adds ddev cronjob ...

### Fixed Issues (if relevant)

1. See OpenMage/magento-lts#3780

(have to run `ddev poweroff` or `ddev restart`)

@ADDISON74 pls test without `ddev get ddev/ddev-cron`! Files in place should work.

### CODE REVIEW

File: .ddev/web-build/openmage.cron
Comment (ADDISON74) [inline .ddev/web-build/openmage.cron:1]: A crontab file must end with EOF. Please insert a newline. Without it you will get the error bellow: 

```php
#20 [web 10/10] RUN { cat /etc/cron.d/*.cron; } | crontab -u ubuntu -
#20 0.349 new crontab file is missing newline before EOF, can't install.
#20 ERROR: process "/bin/bash -c { cat /etc/cron.d/*.cron; } | crontab -u ${username} -" did not complete succes
sfully: exit code: 1
------
 > [web 10/10] RUN { cat /etc/cron.d/*.cron; } | crontab -u ubuntu -:
0.349 new crontab file is missing newline before EOF, can't install.
------
', stderr='failed to solve: process "/bin/bash -c { cat /etc/cron.d/*.cron; } | crontab -u ${username} -" did no
t complete successfully: exit code: 1' 

```

**Review by ADDISON74 — CHANGES_REQUESTED** [review]

**Comment by ADDISON74** [comment]
You basically took the files **Dockerfile.ddev-cron** and **cron.conf** from here https://github.com/ddev/ddev-cron/tree/main/web-build. This means that it is no longer requested to run the ```ddev get ddev/ddev-cron``` command. Then you added openmage.cron file which is responsible for running the OpenMage cron.

Please add the new line in the file **openmage.cron** to approve it.

**Comment by ADDISON74** [comment]
Once approved, we have to create a new PR to change this page https://github.com/OpenMage/magento-lts/blob/main/docs/DDEV.md in the **Setup cronjob** section. Either we remove it completely or we advice the file that needs to be edited openmage.cron to change the default time when the script will be run.

**Comment by sreichel** [comment]
> Please add the new line in the file **openmage.cron** to approve it.

Have you used the file from PR, or from my quick test?

The file from PR has a new line.

**Comment by sreichel** [comment]
> In the DDEV.md file we have to paste these important lines

I'll update docs later, still hope for #3659

**Comment by ADDISON74** [comment]
> > Please add the new line in the file **openmage.cron** to approve it.
> 
> Have you used the file from PR, or from my quick test?
> 
> The file from PR has a new line.
> 
> (also tested cron)

You are right. In PHPStorm that file has two lines. Initially I visually inspected the PR in the browser and GitHub displays only one line.

This PR is OK for being merged.

**Review by ADDISON74 — APPROVED** [review]

**Comment by ADDISON74** [comment]
Wouldn't it be better to use /bin/sh too?

```* * * * * /bin/sh /var/www/html/cron.sh```

**Comment by sreichel** [comment]
What are the benefits? 
Is it required?

Just asking, b/c it works as it is.

**Comment by ADDISON74** [comment]
With this variant ```* * * * * /var/www/html/cron.sh```, the **cron_schedule** table remains empty, it is not filled with data, even after 20 minutes.

When I used this one ```* * * * * /bin/sh /var/www/html/cron.sh```, the table was immediately filled with data.

**Comment by sreichel** [comment]
Ive tested with adding `echo xyz` to cron.sh .... gimm 10 minutes :)

**Comment by ADDISON74** [comment]
Something is happening with the table content. I used the initial variant for the cronjob, running this command ```ddev exec /var/www/html/cron.sh``` should fill the table with data but it's now working. I get into the container ```ddev ssh``` then I run /bin/sh /var/www/html/cron.sh. The table filled with data.

I will check if it is not better */1 * * * *.

**Comment by sreichel** [comment]
Im still waiting ...

1. run version from PR
2. tables got filled (so it should works)
3. truncated table
4. waiting for change again ... zzz

Maybe try highly suggested https://github.com/AOEpeople/Aoe_Scheduler tomorrow.

**Comment by sreichel** [comment]
Okay ... tables are filled again. Sop it works - at least for me/ddev.

**Comment by ADDISON74** [comment]
I will restore openmage.cron to ```* * * * * /var/www/html/cron.sh```, then I will restart DDEV and leave it for a while. Once I see the table is filling up with data I will truncate it and wait again.

**Comment by sreichel** [comment]
Next task: test local mail sending to mailpit/... without extensions ;)

**Comment by sreichel** [comment]
It works for me. Thumbs for morgint.

**Review by ADDISON74 — APPROVED** [review]
It can be approved by another green person to merge it. Adding that important line in the crontab will run the OpenMage tasks.

The other two files used in this PR are from the ddev-cron package, this means that it is no longer requested to install with the ```ddev get ddev/ddev-cron``` command. If the package will be installed, the file will be overwritten with the same content or maybe a new changed one, so there is nothing wrong.

After merging this PR, I will change the DDEV.md file to update the information. I still have problems populating the cron_tab table with data even if the cron is correctly configured. but that will be the subject of a separate discussion.

**Comment by ADDISON74** [comment]
@sreichel - I am deeply disappointed. No one has done anything like this before. I wonder why you created all those PR's anyway? I understand that you have a dispute with two of the maintainers, but what do you have with the rest of us?

**Pull request closed without merging**
